.github/workflows/compliance.yml

```diff
@@ -41,35 +41,21 @@ jobs:
 
     - name: Install python dependencies
       run: |
-        pip3 install setuptools
-        pip3 install wheel
-        pip3 install python-magic junitparser gitlint pylint pykwalify
-        pip3 install west
+        ./scripts/ci/ci_compliance.sh check_compliance python-deps
 
     - name: west setup
       env:
         BASE_REF: ${{ github.base_ref }}
       run: |
-        git config --global user.email "you@example.com"
-        git config --global user.name "Your Name"
-        git remote -v
-        git rebase origin/${BASE_REF}
-        # debug
-        git log  --pretty=oneline | head -n 10
-        west init -l . || true
-        west update 2>&1 1> west.update.log || west update 2>&1 1> west.update2.log
+ 	./scripts/ci/ci_compliance.sh check_compliance west-setup
 
     - name: Run Compliance Tests
       continue-on-error: true
       id: compliance
       env:
         BASE_REF: ${{ github.base_ref }}
       run: |
-        export ZEPHYR_BASE=$PWD
-        # debug
-        ls -la
-        git log  --pretty=oneline | head -n 10
-        ./scripts/ci/check_compliance.py -m Devicetree -m Gitlint -m Identity -m Nits -m pylint -m checkpatch -m Kconfig -c origin/${BASE_REF}..
+        ./scripts/ci/ci_compliance.sh check_compliance compliance-tests
```

Comment:
We really do not need an external script `ci_compliance.sh` to call `check_compliance.py` ...